src/org/recompile/mobile/PlatformPlayer.java

```diff
@@ -103,11 +103,22 @@ public void close()
 
 	public int getState()
 	{
-		if(player.isRunning()==false)
+		try 
 		{
-			state = Player.PREFETCHED;
+			if(player.isRunning()==false)
+			{
+				state = Player.PREFETCHED;
+			}
+			return state;
+		} 
+		catch (Exception e) 
+		{ 
+			System.out.println("Warning: Midlet tried to get the state of a NULL Stream"); 
+			/* Returning a value that isn't 0 to indicate an error, but that's just me
+			assuming it is using standard return codes, i could be wrong. */
+			return 1;  

```

Comment:
Return codes for `javax.microedition.media.Player.getState()` are defined here:
https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/media/Player.html#getState()

The possible states are: `UNREALIZED`, `REALIZED`, `PREFETCHED`, `STARTED`, and `CLOSED`, which you'll find defined in `src/javax/microedition/media/Player.java`  though you should just refer to those and let the compiler take care of putting in the actual value.  They're declared final for a reason. 

Also, are you sure you need to catch an exception there?  There isn't anything that could possibly trigger an exception in the `getState()` method.  The only candidates would be `javax.sound.midi.Sequencer.isRunning` and `javax.sound.sampled.Dataline.isRunning`, but neither one should throw an exception.  (They'll both return false by default.)

The error message is also a bit misleading.  There are places where a null stream might cause a problem, like `midi.setSequence(stream);` on line 217,  but we do catch that exception, we just throw it away and hope for the best.  If that needs to change, that would be the place to do it.  

The relevant Javadocs:
https://docs.oracle.com/javase/7/docs/api/javax/sound/midi/Sequencer.html

https://docs.oracle.com/javase/7/docs/api/javax/sound/sampled/DataLine.html

https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/media/Player.html

Some games actually use exceptions for control flow.  This is a practice so horrid, so vile, so unthinkable that I didn't account for it when I first knocked this together. You'll find a lot of exceptions are caught and then ignored.  Properly identifying and throwing _documented_ exceptions has fixed a few games in the past.  It's worth remembering.

Reply:
Hmm... you're right, maybe that solution isn't the best to the problem some games like Motocross Trial Extreme, Asphalt Nitro faced (nitro needs FileConnect.java though, i'll be redoing the other PR), but then again, i don't think i made it clear what that problem was on my commits, so i apologize.

Basically, it's like this: Those games, and also a few others, try to load up some wavClips as NULL pointers, then try to get the clip's state, which makes FreeJ2ME throw an exception. I'm not sure why they generate NULL wavClips yet, maybe they can't find the file, load it incorrectly, or depend on some improvement on the audio system to work correctly.

Attached an image from AC: Revelations depicting that problem, since it's one of the few that git fixed by that change, and also logged the errors, others like Trial Extreme just stopped crashing with audio enabled, and print the warning i placed.
![image](https://user-images.githubusercontent.com/52329050/173195578-ba4228e1-1871-4a50-b58f-a79df2094e3d.png)

I'm reverting that commit and pushing a new one with a better workaround, since those games appear to fail at the clip's isRunning() call. Hopefully that one won't cause regressions in other games that rely on exceptions for control flow, something that i never expected to be a thing really. I didn't use that check on other functions pertaining to the wavClip because i'd rather not over-engineer things and cause problems by trying to guess if that code would be useful there or not.